### PR TITLE
Processing dm

### PR DESCRIPTION
**Pull request by AtillaTahak**
Hello 👋
In this update:

Built project schema.
Created and ran the necessary migration files.
Included foreign keys.
Have a nice CR. 👍

### CODE REVIEW

File: db/migrate/20220402020159_create_users.rb [hunk 1]
```diff
@@ -1,8 +1,12 @@
 class CreateUsers < ActiveRecord::Migration[7.0]
   def change
     create_table :users do |t|
+      t.string :name
```
Comment (er-contreras) [inline db/migrate/20220402020159_create_users.rb:4]: - [ ] In your migrate directory you have one class called AddColumnToUsers that is adding the same column "name" as you have here. That is causing an error when I run "rails db:migrate". 

Screen Shot 2022-04-13 at 17 41 18

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+end
```
Comment (er-contreras) [inline app/models/user.rb]: - [ ] The user model should have a method that returns the 3 most recent posts. Please add them.

**Review by er-contreras — CHANGES_REQUESTED** [review]
Hi @AtillaTahak, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

almost there

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/models/user.rb [hunk 3]
```diff
@@ -2,4 +2,8 @@ class User < ApplicationRecord
   has_many :posts, foreign_key: :author_id
   has_many :comments, foreign_key: :author_id
   has_many :likes, foreign_key: :author_id
+
+  def load_last_three_posts
+    Post.last(3)
+  end
```
Comment (Emmyn5600) [inline app/models/user.rb:8]: - [ ] Optional suggestion

Use this implementation

posts.order('created_at Desc').limit(3)

File: db/migrate/20220402020159_create_users.rb (hunk 1 above)
Comment (Emmyn5600) [reply db/migrate/20220402020159_create_users.rb:4]: You still have the migration issue.

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @AtillaTahak 

Good job so far! I know you have worked hard on making this milestone. There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
          
## Required Changes :recycle:

- [ ] Check the comment under the review

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!:clap::clap::clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 3 above)
Comment (AtillaTahak) [reply app/models/user.rb:8]: Can you try first rake db:reset after rails db:migrate because when I try everything seems good

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @AtillaTahak 👋🏻, 

Great job making the requested changes 💪🏻!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**